### ISSUE TITLE
[BUG] - Monitoring chart install error throws undefined variable "$forceHTTPSScheme" on Rancher v2.9-head

### ISSUE THREAD

**Issue by izaac** [rancher/rancher#44197]
**Rancher Server Setup**
- Rancher version: `v2.9-12467fadeb539d35a249702dc0933bd84a166433-head`
- Installation option (Docker install/Helm Chart): helm chart
   -  RKE2 - `v1.24.4+rke2r1` and other on `v1.25.9+rke2r1`
- Proxy/Cert Details: valid certs

**Information about the Cluster**
- Kubernetes version: local cluster check kubernetes versions above.
- Cluster Type (Local/Downstream): local / imported.

**User Information**
- What is the role of the user logged in? admin

**Describe the bug**

An error is displayed on the helm installation shell container during the monitoring chart installation.

`Error: parse error at (rancher-monitoring/charts/rke2Scheduler/templates/_helpers.tpl:131): undefined variable "$forceHTTPSScheme"`

**To Reproduce**

- Install a Rancher v2.9-head on an RKE2 local/imported cluster.
- Login as Admin and navigate to Applications and proceed to install the latest Monitoring chart (at the time of this writing `104.0.2+up45.31.1` and just hit install with all the default options. _I had a volume too for persistent prometheus storage if that's relevant_)
- Observe the shell UI component for the helm output logs (easier and automatic on UI) or inspect the helm install container with kubectl.

**Result**

The `rancher-monitoring-crd` app is deployed but the `rancher-monitoring` app fails with a template error:

```
---------------------------------------------------------------------
SUCCESS: helm upgrade --install=true --namespace=cattle-monitoring-system --timeout=10m0s --values=/home/shell/helm/values-rancher-monitoring-crd-104.0.2-up45.31.1.yaml --version=104.0.2+up45.31.1 --wait=true rancher-monitoring-crd /home/shell/helm/rancher-monitoring-crd-104.0.2-up45.31.1.tgz
---------------------------------------------------------------------
helm upgrade --install=true --namespace=cattle-monitoring-system --timeout=10m0s --values=/home/shell/helm/values-rancher-monitoring-104.0.2-up45.31.1.yaml --version=104.0.2+up45.31.1 --wait=true rancher-monitoring /home/shell/helm/rancher-monitoring-104.0.2-up45.31.1.tgz
Release "rancher-monitoring" does not exist. Installing it now.
Error: parse error at (rancher-monitoring/charts/rke2Scheduler/templates/_helpers.tpl:131): undefined variable "$forceHTTPSScheme"
```

**Expected Result**:

Both `rancher-monitoring-crd` and `rancher-monitoring` kubernetes apps of the Rancher Monitoring App should get installed on the local cluster. Attempting to install again should show `Upgrade` not `Install` buttons on the UI

**Screenshots**

**Additional context**

**Comment by skanakal** [comment]
Also, failin on these files...

It appears that a variable $setHTTPSScheme based on .Values.clients.https.forceHTTPSScheme, but then later in the code, using $forceHTTPSScheme, which hasn't been defined.... We need to either change the variable name...

https://github.com/rancher/charts/blob/dev-v2.9/charts/rancher-monitoring/104.0.2%2Bup45.31.1/charts/rke2Scheduler/templates/_helpers.tpl#L105

```
rade.go:524: [debug] copying values from rancher-monitoring (v3) to new release.
Error: UPGRADE FAILED: parse error at (rancher-monitoring/charts/rke2Proxy/templates/_helpers.tpl:134): undefined variable "$forceHTTPSScheme"
helm.go:84: [debug] parse error at (rancher-monitoring/charts/rke2Proxy/templates/_helpers.tpl:134): undefined variable "$forceHTTPSScheme"
UPGRADE FAILED


upgrade.go:524: [debug] copying values from rancher-monitoring (v3) to new release.
Error: UPGRADE FAILED: parse error at (rancher-monitoring/charts/rke2IngressNginx/templates/_helpers.tpl:134): undefined variable "$forceHTTPSScheme"
helm.go:84: [debug] parse error at (rancher-monitoring/charts/rke2IngressNginx/templates/_helpers.tpl:134): undefined variable "$forceHTTPSScheme"
UPGRADE FAILED


upgrade.go:142: [debug] preparing upgrade for rancher-monitoring
upgrade.go:524: [debug] copying values from rancher-monitoring (v3) to new release.
Error: UPGRADE FAILED: parse error at (rancher-monitoring/charts/rke2Etcd/templates/_helpers.tpl:134): undefined variable "$forceHTTPSScheme"
helm.go:84: [debug] parse error at (rancher-monitoring/charts/rke2Etcd/templates/_helpers.tpl:134): undefined variable "$forceHTTPSScheme"


upgrade.go:524: [debug] copying values from rancher-monitoring (v3) to new release.
Error: UPGRADE FAILED: parse error at (rancher-monitoring/charts/rke2ControllerManager/templates/_helpers.tpl:134): undefined variable "$forceHTTPSScheme"
helm.go:84: [debug] parse error at (rancher-monitoring/charts/rke2ControllerManager/templates/_helpers.tpl:134): undefined variable "$forceHTTPSScheme"
UPGRADE FAILED
```

The files needs be to updated:

```
charts/rkeIngressNginx/templates/_helpers.tpl
charts/rkeProxy/templates/_helpers.tpl
charts/rke2IngressNginx/templates/_helpers.tpl
charts/rke2Etcd/templates/_helpers.tpl
charts/rkeControllerManager/templates/_helpers.tpl
charts/kubeAdmProxy/templates/_helpers.tpl
charts/rke2Proxy/templates/_helpers.tpl
charts/rkeScheduler/templates/_helpers.tpl
charts/kubeAdmEtcd/templates/_helpers.tpl
charts/k3sServer/templates/_helpers.tpl
charts/hardenedNodeExporter/templates/_helpers.tpl
charts/kubeAdmControllerManager/templates/_helpers.tpl
charts/rkeEtcd/templates/_helpers.tpl
charts/rke2ControllerManager/templates/_helpers.tpl
charts/kubeAdmScheduler/templates/_helpers.tpl
charts/hardenedKubelet/templates/_helpers.tpl
charts/rke2Scheduler/templates/_helpers.tpl
```

**Comment by joshmeranda** [comment]
Yeah line got changed at some point and didn't notice when rebasing branch. Fix is already implemented, and will be merge with PRl 3317

**Comment by pankajsqe** [comment]
I test this fix on rancher dev-v2.9 and its retest failed.
Error: parse error at (rancher-monitoring/charts/k3sServer/templates/_helpers.tpl:131): undefined variable "$forceHTTPSScheme"
rancher : v2.9-head
k8s : 	v1.26.4+k3s1
Charts: Monitoring (104.0.2+up45.31.1)
<img width="1511" alt="Screenshot 2024-02-01 at 8 26 44 PM" src="https://github.com/rancher/rancher/assets/149133120/aa4474fa-5c5b-459f-a21f-58bc8a0f39cc">

**Comment by pankajsqe** [comment]
Retest passed on Rancher v2.9-head. Monitoring chart was successfully installed on rancher v2.9-head.
Charts: Monitoring (104.0.2+up45.31.1)

<img width="1512" alt="Screenshot 2024-02-05 at 6 53 02 PM" src="https://github.com/rancher/rancher/assets/149133120/2bd197f0-9eed-4107-aad7-f2d7f64d0a6a">


<img width="1509" alt="Screenshot 2024-02-05 at 6 52 50 PM" src="https://github.com/rancher/rancher/assets/149133120/60d218d4-5301-4d87-a8f7-dc6b4403a766">
